main.c

```diff
@@ -299,13 +299,23 @@ libusb_device_handle * LIBUSB_CALL open_device_with_vid(
 					directory = INSTALL_PREFIX "/share/rpiboot/mass-storage-gadget64/";
 					use_bootfiles = 1;
 					snprintf(bootfiles_path, sizeof(bootfiles_path),"%s/%s", directory, "bootfiles.bin");
-					printf("2712: Directory not specified using default %s\n", directory);
+					printf("2712: Directory not specified trying default %s\n", directory);
 				}
 
 				fp_second_stage = check_file(directory, second_stage, 1);
+				if (!fp_second_stage && bcm2712 && use_bootfiles)
```

Comment:
What's the purpose of the use_bootfiles predicate? Indicating that directory was originally unset?

Reply:
directory is an optional argument, but is then overridden, but the code shouldn't search for mass-storage-gadet64 if the -d argument was specified.
I've refactored the code to make it easier to understand